gossip/client/namedtree.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"strings"
+
+	"github.com/quorumcontrol/chaintree/chaintree"
+	"github.com/quorumcontrol/chaintree/nodestore"
+	"github.com/quorumcontrol/messages/v2/build/go/transactions"
+	"github.com/quorumcontrol/tupelo-go-sdk/consensus"
+)
+
+type Generator struct {
+	Namespace string
+	Client    *Client
+}
+
+type NamedChainTree struct {
```

Comment:
As a consumer of this API, why do I want the new type back instead of just a ChainTree ?

Reply:
yeah i agree - it made sense in dgit, but seems odd here